R/demographics-table.R

```diff
@@ -0,0 +1,363 @@
+#' Check if summary function evaluates to the correct format
+#' @param fun The summary function to use for summarizing the continuous data in [pt_demographics()]
+#' @keywords internal
+validate_dem_fun <- function(fun){
+  see <- "See ?pmtables:::dem_cont_fun."
+  form <- formals(fun)
+  assert_that(
+    is.element("...",names(form)),
+    msg = "`fun` must have `...` as a formal argument"
+  )
+  result <- fun(seq(1,5), name = "")
+  assert_that(
+    !is_empty(result),
+    msg = glue("`fun` returned an empty object. {see}")
+  )
+  assert_that(
+    is.data.frame(result),
+    msg = glue("`fun` must return a data frame. {see}")
+  )
+  assert_that(
+    nrow(result)==1,
+    msg = glue("`fun` must return a data frame with a single row. {see}")
+  )
+}
+
+#' Default summary function in pt_demographics for continuous variables
+#'
+#' @param value a vector or column to summarize
+#' @param name the name of the variable currently being summarized
+#' @param ... other arguments passed to `fmt`
+#' @param fmt a function to format the numbers in the continuous data summary
+#' @param digits passed to `fmt`
+#' @param maxex passed to `fmt`
+#'
+#' @return
+#' A tibble with one row and three columns:
+#'  1. `Mean (SD)` the mean and standard deviation of `value`
+#'  2. `Min / Max` the range of the `value`
+#'  3. `Missing` the number of missing `values`
+#'
+#' @details
+#' The summary function should have these arguments:
+#' - `value` the data to summarize
+#' - `name` the name of the data column being summarized; length 1
+#' - `...` absorbs other arguments that might get passed
+#'
+#' These are the important arguments; `name` is not required as a formal
+#' argument but `...` is.  The idea is that the custom summary function
+#' receives the data to be summarized, the name of that data, and other
+#' arguments. Since you are in charge of the function, you can make decisions
+#' about both summaries and the formatting of those summaries inside the
+#' function you are maintaining. Notice that `dem_cont_fun` also contains
+#' arguments `fmt` (a formatting function) and `digits` and `maxex` (arguments
+#' to be passed to `fmt`). This means that you can have your summary function
+#' call the default summary function but with different formatting parameters.
+#'
+#' The summary function should return a data frame, with summary statistics
+#' in the columns.
+#'
+#' @examples
+#' pmtables:::dem_cont_fun(value = seq(1,7), digits = 2)
+#'
+#' @keywords internal
+dem_cont_fun <- function(value = seq(1,5), name = "",  ..., fmt = sig,
```

Comment:
@barrettk The summary function was updated in subsequent development cycles. Is there some change that we should see?

Reply:
I think you just have to run `devtools::build()` and put the new tarball in the directory. Without running `load_all()`, there are a few functions that are outdated. I dont think that will matter once you make a release, but may be worth updating for validation purposes
```
> library(pmtables)
> dem_cont_fun
Error: object 'dem_cont_fun' not found
> pmtables:::dem_cont_fun
function(value = seq(1,5), digits = 3) {
  tibble(
    `mean (sd)` = paste0(sig(mean(value, na.rm = TRUE), digits), " (", sig(sd(value,na.rm=TRUE),digits), ")"),
    `min-max` = paste0(sig(range(value,na.rm=TRUE), digits), collapse = " - ")
  )
}
```

Reply:
@barrettk There is no tarball in the directory. Did you build and install the package on the branch?

Reply:
You can do this on the command line with `make install`

Reply:
Or `R CMD install`

Reply:
Yeah I did that too and there wasn't any issues. I assumed the tarball may have been used for validation purposes because I noticed a script that did so in the mrgsolve package. Everything else looked good though